### PR TITLE
Issue #5877: The theme_file_formatter_table() doesn't respect singular/plural forms

### LINKED ISSUE #5877: The theme_file_formatter_table() doesn't respect singular/plural forms
## Description of the bug

The theme_file_formatter_table() always outputs a singular form of the title "Attachment" regardless if just one or more files are uploaded.

## Steps To Reproduce

To reproduce the behavior:
1. Go to any content type, add a file field and on the _Manage Display_ tab set _Table of files_ format for the field.

Screenshot 2022-12-05 at 7 29 40 PM

2. Create a node of the content type, attach one file, submit the node and you will see a nice table rendered with the header title "Attachment" for the single file.

3. Now, edit the same node or create new one this time uploading two or more files and you will see the singular form of the title doesn't change.

## Expected behavior

Would be nice to have the title take singular or plural forms depending on the number of the files uploaded like:

Screenshot 2022-12-05 at 7 33 25 PM

Screenshot 2022-12-05 at 7 33 31 PM

### PR DESCRIPTION
**Pull request by alanmels**
Fixes https://github.com/backdrop/backdrop-issues/issues/5877

### CODE REVIEW

File: core/modules/file/file.theme.inc [hunk 1]
```diff
@@ -323,14 +323,15 @@ function theme_file_upload_help($variables) {
...
 function theme_file_formatter_table($variables) {
-  $header = array(t('Attachment'), t('Size'));
   $rows = array();
   foreach ($variables['items'] as $delta => $item) {
     $rows[] = array(
       theme('file_link', array('file' => (object) $item)),
       format_size($item['filesize']),
     );
   }
+  count($rows) > 1 ? $title = t('Attachments') : $title = t('Attachment');
```
Comment (avpaderno) [inline core/modules/file/file.theme.inc]: The code should use [`format_plural()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/format_plural/1).

File: core/modules/file/file.theme.inc [hunk 2]
```diff
@@ -330,8 +330,8 @@ function theme_file_formatter_table($variables) {
       format_size($item['filesize']),
     );
   }
-  count($rows) > 1 ? $title = t('Attachments') : $title = t('Attachment');
-  $header = array($title, t('Size'));
+  $title = format_plural(count($rows), 'Attachment', 'Attachments');
+  $header = array(t($title), t('Size'));
```
Comment (avpaderno) [inline core/modules/file/file.theme.inc]: With `format_plural()`, `t()` isn't used. See the code used by [`admin_bar_links_alert()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21admin_bar%21admin_bar.inc/function/admin_bar_links_alert/1), for example.

Comment (klonos) [reply core/modules/file/file.theme.inc]: Good catch @kiamlaluno 👍🏼 ...I gotta remember this.

Comment (klonos) [reply core/modules/file/file.theme.inc]:
```suggestion
  $header = array($title, t('Size'));
```

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

**Comment by klonos** [comment]
Closing this PR, since the issue it was created for was closed as "works as designed". See discussion there for details: https://github.com/backdrop/backdrop-issues/issues/5877

**Pull request closed without merging**